### PR TITLE
Fix Last Respects & Supreme Overlord interaction with reviving

### PR DESCRIPTION
**Pull request by kittenchilly**
Based on this research: https://www.smogon.com/forums/threads/scarlet-violet-battle-mechanics-research.3709545/page-12#post-9408786

### CODE REVIEW

**Comment by kittenchilly** [comment]
Ah thanks. I'll add that then.

**Comment by KrisXV** [comment]
You might want to rebase your branch, there's commits that are already on the server in this pullreq.

File: data/abilities.ts [hunk 1]
```diff
@@ -4283,20 +4283,19 @@ export const Abilities: {[abilityid: string]: AbilityData} = {
...
-			if (faintedAllies) {
+			if (pokemon.side.totalFainted) {
 				this.add('-activate', pokemon, 'ability: Supreme Overlord');
 			}
 		},
 		onModifyAtk(atk, source, target, move) {
-			const faintedAllies = source.side.pokemon.filter(ally => ally.fainted).length;
-			if (faintedAllies < 1) return;
+			const faintedAllies = source.side.totalFainted;
+			if (faintedAllies < 1 || faintedAllies > 5) return;
```
Comment (Karthik99999) [inline data/abilities.ts:4292]:
```suggestion
			const faintedAllies = Math.min(source.side.totalFainted, 5);
			if (faintedAllies < 1) return;
```

File: data/abilities.ts [hunk 2]
```diff
@@ -4283,20 +4283,19 @@ export const Abilities: {[abilityid: string]: AbilityData} = {
...
 			this.debug(`Supreme Overlord atk boost for ${faintedAllies} defeated allies.`);
 			return this.chainModify(1 + (0.1 * faintedAllies));
 		},
 		onModifySpA(spa, source, target, move) {
-			const faintedAllies = source.side.pokemon.filter(ally => ally.fainted).length;
-			if (faintedAllies < 1) return;
+			const faintedAllies = source.side.totalFainted;
+			if (faintedAllies < 1 || faintedAllies > 5) return;
```
Comment (Karthik99999) [inline data/abilities.ts:4298]:
```suggestion
			const faintedAllies = Math.min(source.side.totalFainted, 5);
			if (faintedAllies < 1) return;
```

File: data/moves.ts [hunk 3]
```diff
@@ -10197,7 +10197,7 @@ export const Moves: {[moveid: string]: MoveData} = {
 		accuracy: 100,
 		basePower: 50,
 		basePowerCallback(pokemon, target, move) {
-			return 50 + 50 * pokemon.side.pokemon.filter(ally => ally.fainted).length;
+			return 50 + 50 * pokemon.side.totalFainted;
```
Comment (Karthik99999) [inline data/moves.ts:10200]:
```suggestion
			return 50 + 50 * Math.min(pokemon.side.totalFainted, 100);
```

File: data/abilities.ts [hunk 4]
```diff
@@ -4288,14 +4288,16 @@ export const Abilities: {[abilityid: string]: AbilityData} = {
...
 				this.add('-activate', pokemon, 'ability: Supreme Overlord');
 			}
 		},
 		onBasePower(basePower, source, target, move) {
-			const faintedAllies = Math.min(5, source.side.pokemon.filter(ally => ally.fainted).length);
-			if (faintedAllies) return this.chainModify(1 + (0.1 * faintedAllies));
+			const faintedAllies = Math.min(5, source.side.totalFainted);
+			if (faintedAllies) {
+				this.debug(`Supreme Overlord spa boost for ${faintedAllies} defeated allies.`);
+    	  return this.chainModify(1 + (0.1 * faintedAllies));
```
Comment (KrisXV) [inline data/abilities.ts:4299]:
```suggestion
			if (faintedAllies) {
				this.debug(`Supreme Overlord spa boost for ${faintedAllies} defeated allies.`);
				return this.chainModify(1 + (0.1 * faintedAllies));
```

**Pull request merged**